include/vcpkg/base/expected.h

```diff
@@ -354,6 +354,220 @@ namespace vcpkg
         bool value_is_error;
     };
 
+    template<class Error>
+    struct ExpectedT<void, Error>
```

Comment:
I could see an argument that we should introduce a "do nothing empty type" and ask people to form `ExpectedT<monostate, xxx>` instead.

Reply:
Done.